com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDRaytracingAmbientOcclusion.cs

```diff
@@ -59,103 +54,279 @@ static RTHandle AmbientOcclusionHistoryBufferAllocatorFunction(string viewName,
+        public struct AmbientOcclusionDenoiseResources
+        {
+            // Temporary buffers
+            public RTHandle intermediateBuffer;
+
+            // Output Buffer
+            public RTHandle ambientOcclusionHistory;
+            public RTHandle inputTexture;
+            public RTHandle outputTexture;
+        }
+
+        AmbientOcclusionDenoiseParameters PrepareAmbientOcclusionDenoiseParameters(HDCamera hdCamera, ShaderVariablesRaytracing raytracingCB)
+        {
+            AmbientOcclusionDenoiseParameters rtAOParameters = new AmbientOcclusionDenoiseParameters();
+            var aoSettings = hdCamera.volumeStack.GetComponent<AmbientOcclusion>();
+            rtAOParameters.validRayTracing = m_RenderPipeline.GetRayTracingState();
+            rtAOParameters.denoise = aoSettings.denoise;
+            rtAOParameters.denoiserRadius = aoSettings.denoiserRadius;
+            rtAOParameters.historyValidity = 1.0f;
+#if UNITY_HDRP_DXR_TESTS_DEFINE
+            if (Application.isPlaying)
+                rtAOParameters.historyValidity = 0.0f;
+            else
+#endif
+                // We need to check if something invalidated the history buffers
+                rtAOParameters.historyValidity = m_RenderPipeline.ValidRayTracingHistory(hdCamera) ? 1.0f : 0.0f;
+            rtAOParameters.actualWidth = hdCamera.actualWidth;
+            rtAOParameters.actualHeight = hdCamera.actualHeight;
+            rtAOParameters.viewCount = hdCamera.viewCount;
+            return rtAOParameters;
+        }
+
+        AmbientOcclusionDenoiseResources PrepareAmbientOcclusionDenoiseResources(HDCamera hdCamera, RTHandle inputTexture, RTHandle intermediateTexture, RTHandle outputTexture)
+        {
+            AmbientOcclusionDenoiseResources rtAOResources = new AmbientOcclusionDenoiseResources();
+            rtAOResources.ambientOcclusionHistory = hdCamera.GetCurrentFrameRT((int)HDCameraFrameHistoryType.RaytracedAmbientOcclusion)
+                                                    ?? hdCamera.AllocHistoryFrameRT((int)HDCameraFrameHistoryType.RaytracedAmbientOcclusion, AmbientOcclusionHistoryBufferAllocatorFunction, 1);
+            rtAOResources.inputTexture = inputTexture;
+            rtAOResources.intermediateBuffer = intermediateTexture;
+            rtAOResources.outputTexture = outputTexture;
+            return rtAOResources;
+        }
+
+        // The set of parameters that are used to trace the ambient occlusion
+        public struct AmbientOcclusionComposeParameters
+        {
+            // Generic attributes
+            public bool validRayTracing;
+            public float intensity;
+
+            // Camera data
+            public int actualWidth;
+            public int actualHeight;
+            public int viewCount;
+
+            // Kernels
+            public int intensityKernel;
+
+            // Shaders
+            public ComputeShader aoShaderCS;
+        }
+
+        public struct AmbientOcclusionComposeResources
```

Comment:
In cases where we have only one or two resource parameter like here, it might be better to just pass them directly to the static function as parameters. Otherwise it's a bit of boiler plate code that needs to be duplicated for the render graph version and it  does not really add anything to readability/maintainability.

Reply:
Fixed